nav2_util/include/nav2_util/simple_action_server.hpp

```diff
@@ -54,19 +54,22 @@ class SimpleActionServer
    * @param action_name Name of the action to call
    * @param execute_callback Execution  callback function of Action
    * @param server_timeout Timeout to to react to stop or preemption requests
+   * @param spin_thread  Whether to spin with a dedicated thread internally
    */
+  template<typename NodeT>
   explicit SimpleActionServer(
-    typename nodeT::SharedPtr node,
+    NodeT node,
     const std::string & action_name,
     ExecuteCallback execute_callback,
     CompletionCallback completion_callback = nullptr,
-    std::chrono::milliseconds server_timeout = std::chrono::milliseconds(500))
+    std::chrono::milliseconds server_timeout = std::chrono::milliseconds(500),
+    bool spin_thread = false)
```

Comment:
What happened to the options with the default argument?

Reply:
it's same as the previous one if `spin_thread` is false, in another word, `SimpleActionServer` use default callback group and need spin in `main()`.

Reply:
Yeah, sorry I'm asking why you removed the constructor argument for `options`. I think that was the right thing to do so that someone could override some of the action server's options if they wanted to.

Reply:
yes, i agree with you. the type of `options` is `rcl_action_server_options_t` struct, which contains variables about server qos, etc ([rcl_action/action_server.h#L43-L61](https://github.com/ros2/rcl/blob/414fdbd479b83f2b8af8235ad1fa3aeba2b0027f/rcl_action/include/rcl_action/action_server.h#L43-L61)). i will add it.